src/exo/API_scheduling.py

```diff
@@ -766,6 +792,36 @@ def parse_arg(a):
         return buf_name, args
 
 
+# --------------------------------------------------------------------------- #
+# --------------------------------------------------------------------------- #
+# relevant cursors Dataclass
+
+# TODO: I could not find a way to automatically generate an `__iter__`
+# method for a dataclass. We can easily metaprogram it in `make_cursors_set` below.
+# Another option is to use a `NamedTuple`, but one drawback of this is that we
+# cannot make it a subtype of some base class like we did below.
+
+
+@dataclass
+class CursorsSet:
+    pass
+
+
+def make_cursors_set(sched_op, fields):
```

Comment:
This only appears to be invoked for `simplify()` right now.  Is the implementation only half complete?

Reply:
We didn't want to actually implement the feature until we converge on what the implementation should achieve. Just to avoid redoing work. This was just supposed to be a sample of using this small infrastructure for supporting the feature.